django/db/models/indexes.py

```diff
@@ -31,10 +34,21 @@ def __init__(
             raise ValueError('Index.fields must be a list or tuple.')
         if not isinstance(opclasses, (list, tuple)):
             raise ValueError('Index.opclasses must be a list or tuple.')
+        if not expressions and not fields:
+            raise ValueError('At least one field or expression is required to define an index.')
+        if fields and not all(isinstance(field, str) for field in fields):
+            raise ValueError('Index.fields must only contain strings.')
```

Comment:
This is not strictly related with functional indexes, I'm going to move it to a separate PR.

Reply:
#13883